src/proxy_handler.cr

```diff
@@ -20,14 +20,23 @@ class Mitm::ProxyHandler
 
   def call(context)
     request = context.request
-
     if request.method == "CONNECT"
       host, port = request.resource.split(":", 2)
-
-      context.response.upgrade do |io|
-        magic = io.peek.try &.[1]?
+      counter = 0
+
+      response = context.response
+      response.upgrade do |io|
+        counter += 1
+        Log.trace { "Connect upgrade counter: #{counter}" }
+        if io.closed?
+          Log.trace { "Connect upgrade io closed counter: #{counter}" }
+          next
+        end
+        peek = io.peek
+        next if peek.nil?
```

Comment:
what about adding a trace log in this block?

Reply:
fixed